src/App.tsx

```diff
@@ -15,47 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((f1, f2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return f1.localeCompare(f2);
+      case SortType.LENGTH:
+        return f1.length - f2.length;
+      default:
+        return SortType.NONE;
+    }
+  });
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
+
+  const isResetButtonVisible = isReversed || sortType !== SortType.NONE;
+
+  const sortByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const handleReset = () => {
+    setSortType(SortType.NONE);
+    setReverse(false);
+  };
+
+  const sortByRevers = () => {
```

Comment:
```suggestion
  const handleReverse = () => {
```

please, use meaningfull names